### PR TITLE
feat(prod-sample): deploy module util

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

This adds a util script for deploying modules to `prod-sample`.

## Motivation and Context

To aid testing modules running in production

## How Has This Been Tested?

Locally served modules to `prod-sample`

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "test:lockfile": "lockfile-lint -p package-lock.json -t npm -a npm -o https: -c -i",
     "test:lint": "eslint --ext js,jsx,md,snap .",
     "start": "node lib/server/index.js",
+    "start:inspect": "node --inspect --expose-gc lib/server/index.js",
     "test:unit": "jest --testPathIgnorePatterns integration --config jest.config.js",
```
Comment (mtomcal) [inline package.json:33]: No convenient npm script for this functionality?

File: scripts/deploy-prod-sample-module.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+const deployModuleToProdSampleCDN = async (pathToModule, moduleName) => {
+  const pathToModuleBuildDir = path.resolve(`${pathToModule}/build/`);
+  // use one app git commit sha to namespace modules
+  const gitSha = getGitSha();
+  const pathToOriginModuleStatics = path.resolve(`${nginxOriginStaticsModulesDir}/${gitSha}/${moduleName}`);
+  await fs.ensureDir(pathToOriginModuleStatics);
+  await fs.copy(pathToModuleBuildDir, pathToOriginModuleStatics, { overwrite: true });
+  return pathToOriginModuleStatics;
+};
```
Comment (Francois-Esquire) [inline scripts/deploy-prod-sample-module.js:92]: Would this update while the local production server is running?

File: prod-sample/README.md [hunk 3]
```diff
@@ -57,3 +57,25 @@ local CDN server.
 Similarly, the source code to any
 existing Modules inside the `sample-modules` directory can be modified to then be built
 and bundled from source when `npm run start:prod-sample` is run.
+
+
+## Manually Deploying modules
+
+To aid local production testing use the `deploy-prod-sample-module` script to deploy any module to
+`prod-sample`. This script will create a production build of the module and publish it to the `prod-sample` cdn.
+After it's been published the script will enable to module to be deployed to the `prod-sample` one-app server by
```
Comment (Francois-Esquire) [inline prod-sample/README.md:66]:
```suggestion
After it's been published the script will enable a module to be deployed to the `prod-sample` one-app server by
```

File: scripts/deploy-prod-sample-module.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      envVars: sanitizedEnvVars,
+    });
+  }
+
+  await npmProductionBuild({
+    directory: pathToModule,
+    moduleName,
+    moduleVersion,
+    envVars: sanitizedEnvVars,
+  });
```
Comment (Francois-Esquire) [inline scripts/deploy-prod-sample-module.js:73]: Is this needed to override the development build created after install?

File: package.json (hunk 1 above)
Comment (JAdshead) [reply package.json:33]: this gives us that `npm run start:inspect` ?

Comment (mtomcal) [reply package.json:33]: `npm run serve:prod-sample-module ../some-test-module`?

File: scripts/deploy-prod-sample-module.js (hunk 4 above)
Comment (JAdshead) [reply scripts/deploy-prod-sample-module.js:73]: yeah this ensures the build is run with `NODE_ENV=production`
https://github.com/americanexpress/one-app/pull/308/files#diff-db5dfce483b0c43f844c909e42e27356R72

File: scripts/deploy-prod-sample-module.js (hunk 2 above)
Comment (JAdshead) [reply scripts/deploy-prod-sample-module.js:92]: yes it will

**Review by mtomcal — APPROVED** [review]

**Review by Francois-Esquire — APPROVED** [review]

**Review by infoxicator — APPROVED** [review]

**Pull request merged**